resources/views/users/profile.blade.php

```diff
@@ -126,6 +126,46 @@
                             {!! Form::close() !!}
                         </div>
                     </div>
+
+                    <div class="tab-pane" id="change-password" role="tabpanel">
+                        <div class="card-body">
+                            {!! Form::open(['route' => ['password.change'], 'method' => 'POST', 'class' => 'form-horizontal form-material']) !!}
+
+                                <div class="form-group">
+                                    {!! Form::label('old_password', trans('profile.old_password', ['class' => 'col-md-12'])) !!}
+
+                                    <div class="col-md-12">
+                                        {!! Form::password('old_password', ['class' => ['form-control form-control-line'], 'required' => 'required', 'id' => 'password', 'autofocus' => 'autofocus']) !!}
+
+                                    </div>
+                                </div>
+
+                                <div class="form-group">
+                                    {!! Form::label('password', trans('messages.password', ['class' => 'col-md-12'])) !!}
+
+                                    <div class="col-md-12">
+                                        {!! Form::password('password', ['class' => ['form-control form-control-line'], 'required' => 'required', 'id' => 'password', 'autofocus' => 'autofocus']) !!}
+
+                                    </div>
+                                </div>
+
+                                <div class="form-group">
+                                    {!! Form::label('password_confirmation', trans('profile.confirm_password', ['class' => 'col-md-12'])) !!}
+
+                                    <div class="col-md-12">
+                                        {!! Form::password('password_confirmation', ['class' => ['form-control form-control-line'], 'required' => 'required', 'id' => 'password', 'autofocus' => 'autofocus']) !!}
+
+                                    </div>
+                                </div>
+
+                                <div class="form-group">
+                                    <div class="col-sm-12">
+                                        {{ Form::submit(trans('profile.change_password'), ['class'=>'btn btn-success'])}}
+                                    </div>
+                                </div>
+                            {!! Form::close() !!}
```

Comment:
vì sao lại dùng `{!!` mà không dùng `{{` ?

Reply:
Dùng `{!!` để tránh việc bị tấn công xss nghĩa là hacker có thể chèn các đoạn mã script vào trong `{{`
vd: `{{ <script>arlert("Tan Cong");</script> }}` nó sẽ trở thành `&lt;script&gt;arlert('Tan Cong');&lt;/script&gt;` đoạn mã này trình duyệt có thể hiểu được đó là đoạn script và thực thi nó, để tránh lỗi đó, mình sử dụng `{!! !!}`  nó chỉ xuất ra dữ liệu thô ( `<script>arlert("Tan Cong");</script>` trình duyệt sẽ hiểu nó là 1 đoạn text mà thôi)

Reply:
ngược lại em ơi (facepalm)
nhầm rồi
http://webfaver.com/php-coding/laravel-5/4-laravel-5-view-cach-su-dung-blade-template.html

Reply:
Em lộn a, mình dùng {{ hoặc {!! cái nào cũng được, khi mình lấy value từ database để hiển thị ra view thì mình mới quan tâm nên dùng cái nào ạ

Reply:
vì lý do như trên nên chỉ khi nào cần hiển thị text có khả năng có html mới dùng `{!!` thôi nhé, còn lại sử dụng `{{` cho an toàn

Reply:
Ok chị , vậy giờ fix lại 1 mớ đó luôn a :3